### DISCUSSION TITLE
Cascade `all` vs `all, delete`

### DISCUSSION THREAD

**Discussion by jeffsawatzky** [sqlalchemy/sqlalchemy#6404] [Usage Questions] [answered]
In the documentation [here](https://docs.sqlalchemy.org/en/14/orm/cascades.html#using-foreign-key-on-delete-with-many-to-many-relationships) it says:
>  The all symbol is a synonym for save-update, merge, refresh-expire, expunge, delete

Yet, in examples on that same page it recommends setting cascade=`all, delete` when dealing with cascading deletes.

Since `all` already contains `delete` why do we need to add it again? Is the example wrong? Am I misunderstanding something?

**Comment by zzzeek** [comment, marked as answer]
"delete" is included in "all", and the phrasing of "all, delete" is to indicate that we are talking about the "delete" cascade, but at the same time we don't want users setting ``cascade='delete'`` as that would eliminate all the other cascades, most importantly save-update which is always desirable.

**Reply by jeffsawatzky** [reply]
Thanks, so in practice we only need `cascade="all"` but the extra `, delete` is just there for documentation?

**Reply by zzzeek** [reply]
right now yes.   i agree it is not ideal.
